collect_app/src/main/java/org/odk/collect/android/adapters/TabAdapter.java

```diff
@@ -0,0 +1,57 @@
+package org.odk.collect.android.adapters;
+
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentPagerAdapter;
+
+import org.odk.collect.android.fragments.QRScannerFragment;
+import org.odk.collect.android.fragments.ShowQRCodeFragment;
+
+public class TabAdapter extends FragmentPagerAdapter  {
```

Comment:
We should really be using [ViewPager2](https://developer.android.com/reference/androidx/viewpager2/widget/ViewPager2) and [FragmentStateAdapter](https://developer.android.com/reference/androidx/viewpager2/adapter/FragmentStateAdapter) here. ViewPager was recently deprecated so adding it in now is queuing up rework in the future unfortunately! Would you be able to have a look and see much work it would be to convert these changes over?

Reply:
yep, updated now! thanks!